lightning-invoice/Cargo.toml

```diff
@@ -8,13 +8,18 @@ license = "MIT OR Apache-2.0"
 keywords = [ "lightning", "bitcoin", "invoice", "BOLT11" ]
 readme = "README.md"
 
+[features]
+default = ["std"]
+no-std = ["lightning/no-std"]
+std = ["bitcoin_hashes/std", "num-traits/std", "lightning/std"]
+test_utils = ["lightning/_test_utils"]
+
 [dependencies]
 bech32 = "0.8"
-lightning = { version = "0.0.104", path = "../lightning" }
-secp256k1 = { version = "0.20", features = ["recovery"] }
-num-traits = "0.2.8"
-bitcoin_hashes = "0.10"
+lightning = { version = "0.0.104", path = "../lightning", default-features = false }
+secp256k1 = { version = "0.20", default-features = false, features = ["recovery", "alloc"] }
+num-traits = { version = "0.2.8", default-features = false }
+bitcoin_hashes = { version = "0.10", default-features = false }
 
 [dev-dependencies]
 hex = "0.3"
-lightning = { version = "0.0.104", path = "../lightning", features = ["_test_utils"] }
```

Comment:
leaving this in won't compile with `no-std` because of feature unification

Reply:
It seems like the only clean solution for the feature unification issue is for the test utils to move to a separate crate.

Reply:
Hmmm, can we make `_test_utils` compile with no-std? It may be a substantial diff, so no pressure if not, but ideally our `lightning` tests should work in no-std too.

Reply:
good idea, and it was easy